In [1]:
import pandas as pd
df = pd.read_csv("usarrests.csv")
display(df.shape)
display(df.head(10))
display(df.describe())

(50, 4)

,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91
5,Colorado,7.9,204.0,78
6,Connecticut,3.3,110.0,77
7,Delaware,5.9,238.0,72
8,Florida,15.4,335.0,80
9,Georgia,17.4,NaN,60


,Murder,Assault,UrbanPop
count,50.00000,49.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,130.877435,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


In [5]:
df[df["UrbanPop"] > 100]

,Unnamed: 0,Murder,Assault,UrbanPop
14,Iowa,2.2,56.0,570


In [7]:
df.at[14, 'UrbanPop'] = 57
display(df.describe())

,Murder,Assault,UrbanPop
count,50.00000,50.000000,50.000000
mean,7.78800,181.720000,63.940000
std,4.35551,129.576554,16.453286
min,0.80000,45.000000,6.000000
25%,4.07500,109.000000,53.250000
50%,7.25000,159.000000,66.000000
75%,11.25000,249.000000,76.500000
max,17.40000,879.000000,91.000000


In [2]:
df.isna().sum()

Unnamed: 0    0
Murder        0
Assault       1
UrbanPop      0
dtype: int64

In [3]:
fill_value = df["Assault"].median()
df["Assault"] = df["Assault"].fillna(fill_value)
print(fill_value)

159.0


In [4]:
df.isna().sum()

Unnamed: 0    0
Murder        0
Assault       0
UrbanPop      0
dtype: int64

## Cleaning Log

50 datapoints loaded;
The Assualt data for Georgia loaded with the median of 159;
Changed the data for Iowa's Urban Pop from 570 as that is an impossiblity to 57; as it appears that it may have been a typo and a 0 was added at the end.

https://github.com/skeown13/cs82a-data-science-portfolio